spec/requests/broadcast_announcements_spec.rb

```diff
@@ -0,0 +1,124 @@
+      get edit_broadcast_announcement_url(announcement)
+      expect(response).to be_successful
+    end
+  end
+
+  describe "POST /create" do
+    context "with valid parameters" do
+      it "creates a new BroadcastAnnouncement" do
+        expect {
+          post broadcast_announcements_url, params: {broadcast_announcement: valid_attributes}
+        }.to change(BroadcastAnnouncement, :count).by(1)
+      end
+
+      it "redirects after create" do
+        post broadcast_announcements_url, params: {broadcast_announcement: valid_attributes}
+        expect(response).to have_http_status(:redirect)
+      end
+    end
+
+    context "with invalid parameters" do
+      it "does not create a new BroadcastAnnouncement" do
+        expect {
+          post broadcast_announcements_url, params: {broadcast_announcement: invalid_attributes}
+        }.to change(BroadcastAnnouncement, :count).by(0)
+      end
+
+      it "does not render a successful response" do
+        post broadcast_announcements_url, params: {broadcast_announcement: invalid_attributes}
+        expect(response).not_to be_successful
+      end
+    end
+  end
+
+  describe "PATCH /update" do
+    context "with valid parameters" do
+      let(:new_attributes) {
+        {
+          expiry: Time.zone.yesterday,
+          link: "http://google.com",
+          message: "new_test",
+          user_id: 1,
+          organization_id: 1
+        }
+      }
+
+      it "updates the requested announcement" do
+        announcement = BroadcastAnnouncement.create! valid_attributes
+        patch broadcast_announcement_url(announcement), params: {broadcast_announcement: new_attributes}
+        announcement.reload
+        expect(announcement.message).to eq("new_test")
+      end
+
+      it "redirects after update" do
+        announcement = BroadcastAnnouncement.create! valid_attributes
+        patch broadcast_announcement_url(announcement), params: {broadcast_announcement: new_attributes}
+        announcement.reload
+        expect(response).to have_http_status(:redirect)
+      end
+    end
+
+    context "with invalid parameters" do
+      it "does not render a successful response" do
+        announcement = BroadcastAnnouncement.create! valid_attributes
+        patch broadcast_announcement_url(announcement), params: {broadcast_announcement: invalid_attributes}
+        expect(response).not_to be_successful
+      end
+    end
+  end
+
+  describe "DELETE /destroy" do
+    it "destroys the requested announcement" do
+      announcement = BroadcastAnnouncement.create! valid_attributes
+      expect {
+        delete broadcast_announcement_url(announcement)
+      }.to change(BroadcastAnnouncement, :count).by(-1)
+    end
+
+    it "redirects after delete" do
+      announcement = BroadcastAnnouncement.create! valid_attributes
+      delete broadcast_announcement_url(announcement)
+      expect(response).to have_http_status(:redirect)
```

Comment:
It's OK to have multiple expectations on the same test especially for request / system tests, which take a long time to run. If you're running the exact same code, just put your expectations together.

Reply:
That makes sense. Thank you again for taking the time to explain all this to me